### PR TITLE
Legacy support

### PR DESCRIPTION
**Pull request by pietrushnic**
Not fully working yet. coreboot version is correctly detected. Despite tooolchain is in container it is not detected I think you faced something similar.

### CODE REVIEW

**Comment by pietrushnic** [comment]
@miczyg1  `dev-build` and `release` works for me.  Of course, this will work for builds >=4.0.17 since we need builder support. I will check `release-CI` and let you know, but you can review changes.

File: Dockerfile.legacy [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	cd coreboot && \
+	git checkout pce-fw-builder-legacy && \
+	make crossgcc-i386 CPUS=$(nproc) && \
+	cd /root && \
+	rm -rf coreboot
+
+RUN mkdir /home/coreboot/.ccache && \
+	chown coreboot:coreboot /home/coreboot/.ccache && \
+	echo "export PATH=$PATH:/opt/xgcc/bin" >> /home/coreboot/.bashrc
+
```
Comment (miczyg1) [inline Dockerfile.legacy]: @pietrushnic consider creating /opt/xgcc directory to avoid errors with permissions

Comment (pietrushnic) [reply Dockerfile.legacy]: @miczyg1 this directory is created by coreboot building scripts

**Comment by pietrushnic** [comment]
@miczyg1 pipline seem to pass now

File: Dockerfile.legacy (hunk 1 above)
Comment (miczyg1) [reply Dockerfile.legacy]: @pietrushnic fine. I mentioned it because I saw some problems in the pipeline before.

File: scripts/pce-fw-builder.sh [hunk 2]
```diff
@@ -1,9 +1,10 @@
 #!/bin/bash -x
 
-echo $*
-
 cd /home/coreboot/coreboot
 
+legacy=$1
```
Comment (miczyg1) [inline scripts/pce-fw-builder.sh]: Has variable legacy any use in this script?

File: build.sh [hunk 3]
```diff
@@ -37,19 +75,34 @@ elif [ "$1" == "release" ]; then
...
     VERSION=$1
     OUT_FILE_NAME="$2_${VERSION}.rom"
 
     # remove tag|branch from options
     shift
-    docker run --rm -it -v $PWD/release/coreboot:/home/coreboot/coreboot  \
-        -v $PWD/scripts:/home/coreboot/scripts pcengines/pce-fw-builder:latest \
-        /home/coreboot/scripts/pce-fw-builder.sh $*
 
+    if [ "$legacy" == 1 ]; then
```
Comment (miczyg1) [inline build.sh]: What about legacy branches? They are not tagged so they will not be distinguished by `check_if_legacy` function

File: scripts/pce-fw-builder.sh (hunk 2 above)
Comment (pietrushnic) [reply scripts/pce-fw-builder.sh]: @miczyg1 it is not used now, but I would like to keep because I think there will be customizations soon.

**Comment by miczyg1** [comment]
@pietrushnic resolve the discussions here: https://github.com/pcengines/coreboot/pull/134#pullrequestreview-119002122
and I'm fine with merging after testing pipeline from branch

**Pull request merged**